fix(ViewContainer) removeChild called with null parent

divdavem opened the pull request:
This pull request contains a fix for #997.

In `view_container.js`, `this.templateElement.parentNode` can be null when two template tags are nested in one another. I replaced the call to `DOM.removeChild` by a call to `DOM.remove` which automatically finds the right parent node.

mhevery commented on modules/angular2/src/core/compiler/view_container.js:
```diff
@@ -173,9 +173,9 @@ export class ViewContainer {
   }
 
-  static removeViewNodesFromParent(parent, view) {
+  static removeViewNodes(view) {
```
@vicb @rkirov Doesn't this have perf implication? since now we have to traverse to parent each time. I think there should be a better/faster fix. Can you suggest some?

vicb replied:
Something like

```js
var len = view.nodes.length;
if (len == 0) return;
var parent = view.nodes[0].parentNode;
for (var i = view.nodes.length - 1; i >= 0; --i) {
  DOM.removeChild(parent, view.nodes[i]);
}
```

divdavem committed:
fix(ViewContainer) removeChild called with null parent

In view_container.js, templateElement.parentNode can be null
when two template tags are nested in one another.
Accessing the parent node through view.nodes[0].parentNode fixes
the problem.

closes #997

divdavem commented:
@mhevery @vicb Thank you for your review. I have updated the code.
